spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/metadata/MongoDbMetadataStore.java

```diff
@@ -0,0 +1,197 @@
+				: DEFAULT_COLLECTION_NAME;
+	}
+	
+	
+	/**
+	 * Method put.
+	 * @param key String
+	 * @param value String
+	 * @see org.springframework.integration.metadata.MetadataStore#put(String, String)
+	 */
+	@Override
+	public void put(String key, String value) {
+		Assert.notNull(key, "'key' must not be null.");
+		Assert.notNull(value, "'value' must not be null.");
+		Map<String,String> fileInfo = new HashMap();
+		fileInfo.put(ID_FIELD, key);
+		fileInfo.put(VALUE, value);
+		template.save(fileInfo, collectionName);
+	}
+
+	/**
+	 * Method get.
+	 * @param key String
+	 * @return String
+	 * @see org.springframework.integration.metadata.MetadataStore#get(String)
+	 */
+	@Override
+	public String get(String key) {
+		Assert.notNull(key, "'key' must not be null.");
+		Query query = new Query(Criteria.where(ID_FIELD).is(key));
+		@SuppressWarnings("unchecked")
+		Map<String,String> result = template.findOne(query, Map.class, collectionName);
+		return result==null?null:result.get(VALUE);
+		
+	}
+
+	/**
+	 * Method remove.
+	 * @param key String
+	 * @return String
+	 * @see org.springframework.integration.metadata.MetadataStore#remove(String)
+	 */
+	@Override
+	public String remove(String key) {
+		Assert.notNull(key, "'key' must not be null.");
+		Query query = new Query(Criteria.where(ID_FIELD).is(key));
+		@SuppressWarnings("unchecked")
+		Map<String,String> result = template.findAndRemove(query, Map.class,collectionName);
+		return result==null?null:result.get(VALUE);
+	}
+
+	/**
+	 * Method putIfAbsent.
+	 * @param key String
+	 * @param value String
+	 * @return String
+	 * @see org.springframework.integration.metadata.ConcurrentMetadataStore#putIfAbsent(String, String)
+	 */
+	@Override
+	public String putIfAbsent(String key, String value) {
+		Assert.notNull(key, "'key' must not be null.");
+		Assert.notNull(value, "'value' must not be null.");
+
+		String result = null;
+		Map<String,String> fileInfo = new HashMap();
+		fileInfo.put(ID_FIELD, key);
+		fileInfo.put(VALUE, value);
+		try {
+			template.insert(fileInfo, collectionName);
+		} catch (DuplicateKeyException e) {
+			if(logger.isDebugEnabled()){
+				logger.debug("DUPLICATE KEY VIOLOATION",e);
+			}
+			result = get(key);
+		}
+		return result;
+	}
+
+	/**
+	 * Method replace.
+	 * @param key String
+	 * @param oldValue String
+	 * @param newValue String
+	 * @return boolean
+	 * @see org.springframework.integration.metadata.ConcurrentMetadataStore#replace(String, String, String)
+	 */
+	@Override
+	public boolean replace(String key, String oldValue, String newValue) {
+		Assert.notNull(key, "'key' must not be null.");
+		Assert.notNull(oldValue, "'oldValue' must not be null.");
+		Assert.notNull(newValue, "'newValue' must not be null.");
+		
+		Query query = new Query(Criteria.where(ID_FIELD).is(key).and(VALUE).is(oldValue));
+		Update update = new Update();
+		update.set(VALUE, newValue);
+		WriteResult result = template.updateFirst(query, update, Map.class,collectionName);
```

Comment:
No reason in the `entityClass` here. We are interested in the `isUpdateOfExisting()` only.